slsim/Deflectors/DeflectorTypes/nfw_cluster.py

```diff
@@ -144,3 +144,11 @@ def magnitude(self, band):
             mag = subhalo.magnitude(band)
             total_flux += 10 ** (-0.4 * mag)
         return -2.5 * np.log10(total_flux)
+
+    @property
+    def subhalo_redshifts(self):
```

Comment:
for subhalos, as their definition is to reside in the same halo, have de facto the same redshift and any slight difference along the line of sight is usually negligible (single sheet approximation is valid).
Having the redshifts is required for line of sight halos.
If you want to have line of sight objects (i.e. the 'source' galaxies also acting as secondary deflectors), then perhaps we have a separate definition for those.